File: Algorithm/BOJ/Go/10870.go
// 10870
package main

import (
	"fmt"
)

type Fibonacci struct {
}

func (f Fibonacci) getFibonacci(n int) int {

	if n >= 3 {
		return f.getFibonacci(n-1) + f.getFibonacci(n-2)
	} else if n == 2 {
		return 1
	} else if n == 1 {
		return 1
	} else {
		return 0
	}
}

func main() {
	var n int
	fmt.Scanln(&n)

	fibonacci := Fibonacci{}
	fmt.Println(fibonacci.getFibonacci(n))
}

File: Algorithm/BOJ/C++/10870.cpp
#include <iostream>

using namespace std;



class Fibonacci
{
public:
	int getFibonacci(int n);
};



int Fibonacci::getFibonacci(int n) {
	
	if ( n >= 3 ) {
		return getFibonacci(n - 1) + getFibonacci(n - 2);
	}
	else if (n == 2) {
		return 1;
	}
	else if (n == 1) {
		return 1;
	}
	else {
		return 0;
	}
	
}


int main() {

	int n;
	cin >> n;

	Fibonacci fibonacci;
	cout << fibonacci.getFibonacci(n);
}
